modules/phase_field/src/postprocessors/GrainTracker.C

```diff
@@ -662,18 +707,49 @@ GrainTracker::remapGrains()
    */
   if (_is_master)
   {
-    mooseAssert(!_unique_grains.empty(), "Unique grains data structure is empty!");
-
-    // Resize our vector to the largest grain index (last one in the map) plus one to hold all of the potential remapping indices
-    std::vector<unsigned int> grain_id_to_existing_var_idx(_unique_grains.rbegin()->first + 1, std::numeric_limits<unsigned int>::max());
-    for (auto & grain_pair : _unique_grains)
+    // Build the map to detect difference in _var_idx mappings after the remap operation
+    std::map<unsigned int, unsigned int> unique_id_to_existing_var_idx;
+    for (auto & grain : _feature_sets)
     {
-      mooseAssert(!(grain_pair.second._status == Status::CLEAR), "Grain " << grain_pair.first << " status in wrong state.");
+      // Unmark the grain so it can be used in the remap loop
+      grain._status = Status::CLEAR;
 
-      if (grain_pair.second._status == Status::MARKED)
+      unique_id_to_existing_var_idx[grain._id] = grain._var_idx;
+    }
+
+    // Make sure that all split pieces of an EBSD grain are on the same OP
+    if (_ebsd_reader)
+    {
+      for (auto i = beginIndex(_feature_sets); i < _feature_sets.size(); ++i)
```

Comment:
Almost! In this loop I actually need the indices in line 735. In the second loop I never need the indices.